src/k2/transport/Payload.cpp

```diff
@@ -409,7 +409,8 @@ Payload Payload::shareRegion(size_t startOffset, size_t nbytes){
     seek(startOffset);
     nbytes = std::min(_size - _currentPosition.offset, nbytes);
 
-    Payload shared(_allocator);
+    BinaryAllocator copied_allocator = _allocator;
+    Payload shared(std::move(copied_allocator));
```

Comment:
I think this could just be
```suggestion
    Payload shared(BinaryAllocator(_allocator));
```

Reply:
done, thanks for suggestions.